robottelo/performance/graph.py

```diff
@@ -0,0 +1,97 @@
+import pygal
+
+
+def generate_bar_chart_stat(stat_dict, head, filename, lengend):
+    """Generate Bar chart for stat on concurrent subscription"""
+    bar_chart = pygal.Bar()
+    bar_chart.title = head
+    bar_chart.x_labels = ('min', 'mean', 'max', 'std')
+    bar_chart.x_title = 'Statistics'
+    bar_chart.y_title = 'Time (s)'
+    for key in range(len(stat_dict)):
+        bar_chart.add(
+            '{0}-{1}'.format(lengend, key),
+            stat_dict.get(key)
+        )
+    bar_chart.render_to_file(filename)
+
+
+def generate_line_chart_bucketized_stat(stat_dict, head, filename,
+                                        bucket_size, num_buckets):
+    """Generate Line chart for stat on concurrent subscription"""
+    if bucket_size == 0:
+        return
+
+    # parameters for formats on charts
+    lengend = ('min', 'median', 'max', 'std')
+    buckets = [
+        '{0}-{1}'.format(
+            bucket_size * i + 1,
+            bucket_size * (i + 1)
+        )
+        for i in range(num_buckets)
+    ]
+    line_chart = pygal.Line()
+    line_chart.title = head
+    line_chart.x_labels = buckets
+    line_chart.x_title = 'Buckets'
+    line_chart.y_title = 'Time (s)'
+    for stat, stat_value in enumerate(lengend):
```

Comment:
Thanks for the explanation.

Beware that `iteritems()` is only available in Python 2, not Python 3. Also, you still have the issue of creating unnecessary variables. I'll post an alternate solution.